src/Registry/TypeRegistry.php

```diff
@@ -774,7 +821,13 @@ public function get_type( string $type_name ) {
 	 * @return array
 	 */
 	public function get_types() {
-		return $this->types;
+
+		// The full map of types is merged with eager types to support the
+		// rename_graphql_type API.
+		//
+		// All of the types are closures, but eager Types are the full
+		// Type definitions up front
+		return array_merge( $this->types, $this->get_eager_type_map() );
```

Comment:
I don't think we need to merge this, since `get_eager_type_map()` calls `get_type()`, which populates `$this->types`.

Seems like `get_eager_type_map` could just be `get_eager_types` and return `void`, since it populates `$this->types`

Reply:
Without the `array_merge()` everything fails 😬 

![Screen Shot 2021-07-23 at 1 23 32 PM](https://user-images.githubusercontent.com/1260765/126831457-72f77a7b-b9a4-4c97-b5f3-65a98125187a.png)

The reason we need this:

`get_eager_type_map()` returns an array of Types that need to be loaded up-front, instead of referencing a callback. GraphQL-PHP needs this for Types that are not referenced directly by a field, but only show as part of a Union or Interface. 

So while `$this->types[]` looks like: 

```php
[
  'RootQuery' => function() {}, // callback
  'RootMutation' => function() {}, // callback 
  'Post' => function() {}, // callback
  ...
]
```

The eager type map looks like: 

```php
[
  'EagerType' => WPObjectType // Full Type Def, not a callback
]
```

So the output of `get_types()` needs to be a mix of the lazy loaded Types that map to a callback (which will load a Type Object) _and_ the eager types that map directly to a Type Object: 

```php
[
  'RootQuery' => function() {},
  'RootMutation' => function() {},
  'Post' => function() {},
  'EagerType' => WPObjectType 
  ...
]
```

Reply:
Yup yup, got it. I read that method incorrectly. Thanks!